Platform/OpenVariableRuntimeDxe/VariableDxe.c

```diff
@@ -685,19 +685,14 @@ VariableServiceInitialize (
   SystemTable->RuntimeServices->SetVariable         = VariableServiceSetVariable;
   //
   // Avoid setting UEFI 2.x interface member on EFI 1.x.
+  // NB It is important to set this when possible: we have completely replaced
+  // SetVariable and GetVariable and QueryVariableInfo must be consistent with them.
+  // We attempt allow for a previous inconsistently faked UEFI 2+.
   //
-  // First test all systable elements as some may have been spoofed and pass a limited element check
-  // Then check that QueryVariableInfo is specifically available before setting the interface member
-  //
-  if (  ((SystemTable->Hdr.Revision >> 16U) > 1)
-     && ((SystemTable->BootServices->Hdr.Revision >> 16U) > 1)
-     && ((SystemTable->RuntimeServices->Hdr.Revision >> 16U) > 1))
+  if (  (SystemTable->RuntimeServices->Hdr.Revision >= EFI_2_00_SYSTEM_TABLE_REVISION)
+     && (SystemTable->RuntimeServices->Hdr.HeaderSize > OFFSET_OF (EFI_RUNTIME_SERVICES, QueryVariableInfo)))
```

Comment:
I believe https://github.com/acidanthera/OpenCorePkg/pull/405 was over-cautious. In fact, the important thing here is defintely to provide QueryVariableInfo if there is space to put it, not to avoid providing it if there is any sign of a previous forge.

Reply:
I have avoided also looking at SystemTable->Hdr.Revision (e.g. perhaps trying to fake if _either_ version is 2 or above and there is enough space to put the pointer), because of the existence of code like this:

https://github.com/tianocore/edk/blob/master/Foundation/Efi/Include/EfiApi.h#L946-L960